app/src/main/java/org/oppia/app/player/audio/AudioFragment.kt

```diff
@@ -5,44 +5,58 @@ import android.os.Bundle
 import android.view.LayoutInflater
 import android.view.View
 import android.view.ViewGroup
+import androidx.fragment.app.Fragment
 import org.oppia.app.fragment.InjectableFragment
 import javax.inject.Inject
 
-private const val TAG_DIALOG = "LANGUAGE_DIALOG"
+private const val TAG_LANGUAGE_DIALOG = "LANGUAGE_DIALOG"
 
-/** Fragment that controls audio for a state and content.*/
-class AudioFragment : InjectableFragment() {
+/** Fragment that controls audio for a content-card. */
+class AudioFragment : InjectableFragment(), LanguageInterface {
   @Inject
   lateinit var audioFragmentPresenter: AudioFragmentPresenter
-
-  lateinit var languageInterface: LanguageInterface
+  private lateinit var languageInterface: LanguageInterface
+  var selectedLanguageCode: String = "en"
 
   override fun onAttach(context: Context?) {
     super.onAttach(context)
     fragmentComponent.inject(this)
   }
 
+  override fun onCreate(savedInstanceState: Bundle?) {
+    super.onCreate(savedInstanceState)
+    onAttachToParentFragment(this)
+  }
+
   override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
     return audioFragmentPresenter.handleCreateView(inflater, container)
   }
 
-  fun languageSelectionClicked() {
-    languageInterface = object : LanguageInterface {
-      override fun onLanguageSelected(currentLanguageCode: String) {
-        audioFragmentPresenter.languageSelected(currentLanguageCode)
-      }
+  private fun onAttachToParentFragment(fragment: Fragment) {
+    try {
+      languageInterface = fragment as LanguageInterface
+    } catch (e: ClassCastException) {
+      throw ClassCastException(
```

Comment:
What's the purpose of this catch & rethrow?

Reply:
This was used from the reference provided in previous conversation https://stackoverflow.com/questions/23142956/sending-data-from-nested-fragments-to-parent-fragment/23144683#23144683

But after checking again, the code works even without using this.

Reply:
This `try-catch-throw` was mainly checking if the casting has been done correctly or not. And after checking the code correctly, we don't need this because our code is using LanguageInterface correctly.

Therefore removing this `try-catch-throw` is safe.